tests/Repositories/TaskRepositoryTest.php

```diff
@@ -94,14 +99,105 @@ public function it_can_retrieve_task_list_of_given_projects()
             'project_id' => $project->id,
         ]);
 
-        $getTask = $this->taskRepo->getTaskList([$project->id]);
+        $taskList = $this->taskRepo->getTaskList([$project->id]);
 
-        $this->assertCount(2, $getTask);
+        $this->assertCount(2, $taskList);
 
-        $taskIds = $getTask->keys();
-        $tasks->map(function (Task $task) use ($getTask, $taskIds) {
-            $this->assertContains($task->title, $getTask);
+        $taskIds = $taskList->keys();
+        $tasks->map(function (Task $task) use ($taskList, $taskIds) {
+            $this->assertContains($task->title, $taskList);
             $this->assertContains($task->id, $taskIds);
         });
     }
+
+    /** @test */
+    public function it_can_add_comment()
+    {
+        $task = factory(Task::class)->create();
+        $comment = $this->taskRepo->addComment([
+            'comment' => 'random text',
+            'task_id' => $task->id,
+        ]);
+
+        $this->assertNotEmpty($comment);
+        $this->assertEquals('random text', $comment->comment);
+        $this->assertEquals($this->defaultUserId, $comment->createdUser->id);
+    }
+
+    /** @test */
+    public function test_can_return_unique_task_index()
+    {
+        $project = factory(Project::class)->create();
+        factory(Task::class)->create([
+            'project_id'  => $project->id,
+            'task_number' => 3,
+        ]);
+        $uniqueIndex = $this->taskRepo->getIndex($project->id);
+
+        $this->assertNotEmpty($uniqueIndex);
+        $this->assertEquals(4, $uniqueIndex, '+1 index');
+    }
+
+    /** @test */
+    public function test_can_return_task_number_one_when_no_tasks_on_given_project()
+    {
+        $project = factory(Project::class)->create();
+
+        $uniqueIndex = $this->taskRepo->getIndex($project->id);
+
+        $this->assertNotEmpty($uniqueIndex);
+        $this->assertEquals(1, $uniqueIndex);
+    }
+
+    /** @test */
+    public function test_can_get_active_tasks_of_logged_in_user_for_given_project_id()
+    {
+        $user = factory(User::class)->create();
```

Comment:
change variable name to real name as we discussed, `$farhan` | `$vishal` or something else, do it everywhere